src/plugins/content_management/public/components/section_render.tsx

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useState, useEffect, useMemo } from 'react';
+import { useObservable } from 'react-use';
+import { BehaviorSubject } from 'rxjs';
+
+import { Content, Section } from '../services';
+import { EmbeddableInput, EmbeddableRenderer, EmbeddableStart } from '../../../embeddable/public';
+import { DashboardContainerInput } from '../../../dashboard/public';
+import { createCardInput, createDashboardInput } from './section_input';
+import { CARD_CONTAINER } from './card_container/card_container';
+import { EuiTitle } from '@elastic/eui';
+import { SavedObjectsClientContract } from 'opensearch-dashboards/public';
+
+interface Props {
+  section: Section;
+  contents$: BehaviorSubject<Content[]>;
+  embeddable: EmbeddableStart;
+  savedObjectsClient: SavedObjectsClientContract;
+}
+
+export interface CardInput extends EmbeddableInput {
+  description: string;
+}
+
+const DashboardSection = ({ section, embeddable, contents$, savedObjectsClient }: Props) => {
+  const contents = useObservable(contents$);
+  const [input, setInput] = useState<DashboardContainerInput>();
+
+  useEffect(() => {
+    if (section.kind === 'dashboard') {
+      createDashboardInput(section, contents ?? [], { savedObjectsClient }).then((ds) =>
+        setInput(ds)
+      );
+    }
+  }, [section, contents]);
+
+  const factory = embeddable.getEmbeddableFactory('dashboard');
+
+  if (section.kind === 'dashboard' && factory && input) {
+    // const input = createDashboardSection(section, contents ?? []);
+    return <EmbeddableRenderer factory={factory} input={input} />;
+  }
+
+  return null;
+};
+
+const CardSection = ({ section, embeddable, contents$ }: Props) => {
+  const contents = useObservable(contents$);
+  const input = useMemo(() => {
+    return createCardInput(section, contents ?? []);
+  }, [section, contents]);
+
+  const factory = embeddable.getEmbeddableFactory(CARD_CONTAINER);
+
+  if (section.kind === 'card' && factory && input) {
+    return (
+      <div style={{ padding: '0 8px' }}>
+        <EuiTitle size="s">
+          <h2>{section.title}</h2>
+        </EuiTitle>
+        <EmbeddableRenderer factory={factory} input={input} />
+      </div>
+    );
+  }
+
+  return null;
+};
+
+const CustomSection = ({ section, contents$ }: Props) => {
+  const contents = useObservable(contents$);
+
+  if (section.kind === 'custom' && contents) {
+    return section.render(contents);
+  }
+
+  return null;
+};
+
+export const SectionRender = (props: Props) => {
+  if (props.section.kind === 'dashboard') {
+    return <DashboardSection {...props} />;
+  }
+
+  if (props.section.kind === 'card') {
+    return <CardSection {...props} />;
+  }
```

Comment:
Nit: is it necessary that we have `card` section?

Reply:
The card is a pretty common UX pattern nowadays, for example, homepage has this, overview pages will have it, so I think it might make sense to have a predefined type of card section.

Reply:
@ashwin-pc Attached a screenshot 
<img width="1029" alt="image" src="https://github.com/user-attachments/assets/de15ff16-a470-4b95-93cf-48f5e8140030">